packages/pwa-kit-runtime/package.json

```diff
@@ -53,7 +53,6 @@
     "internal-lib-build": "^2.0.0-dev.4",
     "nock": "^13.1.1",
     "node-fetch": "2.6.1",
-    "pwa-kit-cli": "^2.0.0-dev.4",
```

Comment:
We need this package though. We are using `pwa-kit-cli` for local dev. See [here](https://github.com/SalesforceCommerceCloud/pwa-kit/blob/v2/packages/pwa-kit-runtime/src/ssr/server/express.js#L490)

Reply:
The linked file (`express.js`) is the only place that `pwa-kit-runtime` uses `pwa-kit-cli`. The imported file (`build-dev-server.js`) is only used by `express.js`. To resolve the circular dependency, I think it makes sense to just move `build-dev-server.js` into `pwa-kit-runtime`, as it's not used by any other packages, not even by `pwa-kit-cli`.

Reply:
@alexvuong It actually isn't being used there. The usage on the line that you called out is covered by the peer dependency [here](https://github.com/SalesforceCommerceCloud/pwa-kit/blob/e86f7a147fd7104f7a675f4ac4443869be8e9492/packages/pwa-kit-runtime/package.json#L63)

Reply:
@wjhsf I had similar thoughts. This would be more of an architectural change to @olibrook work, it would also go beyond fixing the circular dependency as it's not required to fix this issue. He might see this comment and weigh in.

Reply:
@alexvuong We do not need it – if we did, it wouldn't even have been in `devDependencies` before. That's why the code does the dynamic `eval('require')` import.

@bendvc Your suggestion looks good to me if you would like to silence the warnings. And the test strategy looks fine to me, too.

Reply:
@bendvc got it. What @wjhsf said makes sense to me too.

Reply:
@alexvuong @wjhsf I reckon that @olibrook 's intentions where to separate the dev server factory from the runtime package is it doesn't really have anything to do with the runtime per-say. You can imaging that the runtime isn't concerned with how your application is developed, only that can run projects that follow its API. 

For example, you can create a project that doesn't use webpack at all, and therefore doesn't use webpack-dever-server, so it doesn't make sense that we have that code in the runtime. Previously the cli package was called pwa-kit-build, and it sounded more like a package that might have a dev server in it, but now that it's called pwa-kit-cli it seems a little less likely to be a package that contains a dev sever factory. I think that might be the source of the code smell. 

That being said, this is outside of the scope of this ticket I think. The circular dep only existed via a dev dependency, which we have removed. So I think we are good to go.